Sources/Web5/Credentials/PresentationExchange.swift

```diff
@@ -0,0 +1,219 @@
+                vcJwt
+            }
+            if !vcJwtList.isEmpty {
+                result[inputDescriptor] = vcJwtList
+            }
+        }
+        return result
+    }
+ 
+    // MARK: - VC Satisfies Input Descriptor
+    private static func vcSatisfiesInputDescriptor(vc: VCDataModel, inputDescriptor: InputDescriptorV2) throws -> Bool {
+        // If the Input Descriptor has constraints and fields defined, evaluate them.
+        guard let fields = inputDescriptor.constraints.fields else {
+            // If no fields are defined, VC satisfies
+            return true
+        }
+
+        let requiredFields = fields.filter { !($0.optional ?? false) }
+
+        for field in requiredFields {
+            
+            // Takes field.path and queries the vc to see if there is a corresponding path.
+            let vcJson = try JSONEncoder().encode(vc)
+            guard let matchedPathValues = vcJson.query(values: field.path) else { return false }
+            
+            
+            if matchedPathValues.isEmpty {
+                // If no matching fields are found for a required field, the VC does not satisfy this Input Descriptor.
+                return false
+            }
+
+            // If there is a filter, process it
+            if let filter = field.filter {
+                let fieldName = field.path[0]
+                
+                var filterProperties: [String: [String: String]] = [:]
+
+                for (filterKey, filterValue) in filter {
+                    guard let filterRule = filterValue.value as? String else {
+                        return false
+                    }
+                    
+                    filterProperties[fieldName, default: [:]][filterKey] = filterRule
+                }
+                
+                let satisfiesSchemaMatches = try matchedPathValues.filter { value in
+                    let result = try JSONSchema.validate([fieldName: value], schema: [
+                      "properties": filterProperties
+                    ])
+                    return result.valid
+                }
+                
+                if satisfiesSchemaMatches.isEmpty {
+                    // If the field value does not satisfy the schema, the VC does not satisfy this Input Descriptor.
+                    return false
+                }
+            }
+        }
+
+        // If the VC passes all the checks, it satisfies the criteria of the Input Descriptor.
+        return true
+    }
+
+}
+
+// MARK: - Errors
+extension PresentationExchange {
+    public enum Error: LocalizedError {
+        case missingCredentialObject
+        case missingCredentials(Int, Int)
+        
+        public var errorDescription: String? {
+            switch self {
+            case .missingCredentialObject:
+                return "Failed to find Verifiable Credential object in parsed JWT"
+            case .missingCredentials(let totalNeeded, let actualReceived):
```

Comment:
`missingCredentials` feels a bit odd to me because it seems like the error message is actually about missing input descriptors? is this error name standardized across our other sdks?